actions/class.RestQtiTests.php

```diff
@@ -0,0 +1,179 @@
+		 "itemsList"=> RDFS_ITEMSLIST
+		   
+	    ));
+	}
+	
+	
+		protected function importQtiPackage($parameters){
+		$report=null;
+		if(isset($_FILES["qtiPackage"]))
+		{
+			if($_FILES["qtiPackage"]["name"]) {
+				$filename = $_FILES["qtiPackage"]["name"];
+				$source = $_FILES["qtiPackage"]["tmp_name"];
+				$type = $_FILES["qtiPackage"]["type"];
+
+				$name = explode(".", $filename);
+				$accepted_types = array('application/zip', 'application/x-zip-compressed', 'multipart/x-zip', 'application/x-compressed');
+				foreach ($accepted_types as $mime_type) {
+					if ($mime_type == $type) {
+						$okay = true;
+						break;
+					}
+				}
+				$continue = strtolower($name[1]) == 'zip' ? true : false;
+				if (!$continue) {
+					throw new common_exception_InvalidArgumentType();
+
+				}
+				$path = dirname(__FILE__) . '/';  // absolute path to the directory where zipper.php is in
+				$filenoext = basename($filename, '.zip');  // absolute path to the directory where zipper.php is in (lowercase)
+				$filenoext = basename($filenoext, '.ZIP');  // absolute path to the directory where zipper.php is in (when uppercase)
+
+				$targetdir = $path . $filenoext; // target directory
+				$targetzip = $path . $filename; // target zip file
+
+				/* create directory if not exists', otherwise overwrite */
+				/* target directory is same as filename without extension */
+
+				if (is_dir($targetdir)) rmdir_recursive($targetdir);
+				if(move_uploaded_file($source, $targetzip)) {
+
+					$report = $this->service->importFromArray($parameters, $targetzip);
+				}
+			}else{
+				throw new common_exception_InvalidArgumentType();
+			}
+
+
+
+		}
+		return $report;
+	}
+	/**
+	 * Optionnal Requirements for parameters to be sent on every service
+	 *
+	 */
+	protected function getParametersRequirements()
+	{
+	    return array(
+		/** you may use either the alias or the uri, if the parameter identifier
+		 *  is set it will become mandatory for the method/operation in $key
+		* Default Parameters Requirents are applied
+		* type by default is not required and the root class type is applied
+         *    @example :"post"=> array("login", "password")
+		*/
+		"post"=> array("qtiPackage"),
+		"itemsList"=> RDFS_ITEMSLIST
+		
+	    );
+	}
+	
+	/**
+	 *
+	 * @author Rashid and Absar -PCG Team
+	 */
+	 
+
+
+	protected function post() {
+		try {
+			$parameters = $this->getParameters();
+			/**
+			 * This code snippet  import QTI Package or create new item
+			 *
+			 * @author Rashid Mumtaz & Absar - PCG Team - {rashid.mumtaz372@gmail.com}
+			 * @license GPLv2  http://www.opensource.org/licenses/gpl-2.0.php
+			 * @package taoItems
+
+			 *
+			 */
+			if(isset($parameters[TAO_QTITEST_QTIPACKAGE_PROPERTY])){
+				// Import QTI Package
+				
+				
+				 $data=$this->importQtiPackage($parameters);
+				 
+			
+				 
+				 
+				 $reflectionObject = new ReflectionObject($data);
+				 $property = $reflectionObject->getProperty('elements');
+				 
```

Comment:
Reflection should not be required here, as report implements iterator, you can iterate over it:

foreach ($data as $subreport) {
    $subreport->getData();
}